app/services/security/cspHeader.server.ts

```diff
@@ -0,0 +1,54 @@
+import { serverConfig } from "~/config/config.server";
+
+export const getCspHeader = (args: {
+  nonce: string;
+  environment: string;
+  additionalConnectSrc?: string[];
+}) => {
+  const directives: Record<string, string[]> = {
+    "default-src": ["'self'"],
+    "script-src": [
+      "'self'",
+      "https:",
+      `'nonce-${args.nonce}'`,
+      "'strict-dynamic'",
```

Comment:
Nit: Let's add a little comment here...
```suggestion
      "'strict-dynamic'", // allows/favors hashes & nonces over domain host lists
```